### PR TITLE
[FEATURE ]First try on memory efficient streaming/checking for mzmls in SE adapter

### PR DESCRIPTION
**Pull request by jpfeuffer**

### CODE REVIEW

File: src/openms/source/FORMAT/MascotGenericFile.cpp [hunk 1]
```diff
@@ -318,10 +318,8 @@ namespace OpenMS
     }
     if (spec.size() >= 10000)
     {
-      String msg = "Spectrum to be written as MGF has " + String(spec.size()) +
-        " peaks; the upper limit is 10,000. Only centroided data is allowed - this is most likely profile data.";
-      throw Exception::IllegalArgument(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION,
-                                       msg);
+      OPENMS_LOG_WARN << "Spectrum to be written as MGF has " + String(spec.size()) +
```
Comment (cbielow) [inline src/openms/source/FORMAT/MascotGenericFile.cpp:321]: MGF file format actually mandates less than 10k peaks,
see http://www.matrixscience.com/help/data_file_help.html -->
```
There is an upper limit of 10,000 peaks per individual MS/MS spectrum.
If you see an error message reporting that this limit has been exceeded,
 it almost certainly means that your data are profile data, and not peak lists.
 It is very unlikely that a single MS/MS spectrum could ever contain more than 
1000 genuine peaks, never mind 10,000.```

File: src/openms/source/FORMAT/MzMLFile.cpp [hunk 2]
```diff
@@ -252,4 +253,32 @@ namespace OpenMS
     consumer->setExperimentalSettings(experimental_settings);
   }
 
+  std::map<UInt,std::pair<Size,Size>> MzMLFile::getCentroidInfo(const String& filename)
+  {
+    bool oldoption = options_.getFillData();
+    options_.setFillData(false);
+    MSDataTransformingConsumer* c = new MSDataTransformingConsumer();
```
Comment (cbielow) [inline src/openms/source/FORMAT/MzMLFile.cpp:260]: Why create on heap?

File: src/topp/CometAdapter.cpp [hunk 3]
```diff
@@ -627,42 +628,38 @@ class TOPPCometAdapter :
...
 
-    PeakMap exp;
-    MzMLFile mzml_file;
-    mzml_file.getOptions().setFillData(false); // only load metadata for spectra
-    mzml_file.getOptions().setMSLevels({2, 3}); // only load MS2 and MS3
-    mzml_file.setLogType(log_type_);
-    mzml_file.load(inputfile_name, exp);
+    int ms_level = getIntOption_("ms_level");
+    const auto& centroid_info = MzMLFile().getCentroidInfo(inputfile_name);
+    std::cout << centroid_info.size() << std::endl;
```
Comment (cbielow) [inline src/topp/CometAdapter.cpp:633]: remove cout message
?!

File: src/openms/source/FORMAT/MascotGenericFile.cpp (hunk 1 above)
Comment (jpfeuffer) [reply src/openms/source/FORMAT/MascotGenericFile.cpp:321]: see https://github.com/OpenMS/OpenMS/issues/4655

Comment (jpfeuffer) [reply src/openms/source/FORMAT/MascotGenericFile.cpp:321]: I can revert it, but IMHO this is just plain arbitrary.

Comment (cbielow) [reply src/openms/source/FORMAT/MascotGenericFile.cpp:321]: but its their format...

and 10k peaks in an MS2 spectrum... this has to be wrong...

**Comment by jpfeuffer** [comment]
http://cdash.openms.de/testDetails.php?test=91827267&build=443403

False positive? Can't see anything wrong. Codacy also has cppcheck and does not complain.

**Pull request merged**